optimum/intel/ipex/modeling_base.py

```diff
@@ -693,3 +718,16 @@ def _ipex_reorder_cache(
             tuple(past_state.index_select(0, beam_idx.to(past_state.device)) for past_state in layer_past)
             for layer_past in past_key_values
         )
+
+
+def _ipex_crop_past_key_values(model, past_key_values, max_length):
+    if isinstance(model, IPEXModel):
+        new_past_key_values = []
+        for i in range(len(past_key_values)):
+            pkv = []
+            pkv.append(past_key_values[i][0][:, :max_length, :max_length, :])
+            pkv += [past_key_values[i][_] for _ in range(1, 4)]
```

Comment:
why cropping the first one only ?

Reply:
Because the the real past_key_values are saved in past_key_values[i][1] and past_key_values[i][2] with a fixed_length (pre allocate all memories). So the real length are saved in past_key_values[i][0], the OP will select past key values by past_key_values[i][1][:past_key_values[i][0].shape[2]] and past_key_values[i][2][:past_key_values[i][0].shape[2]]. For more details, please see [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache).

Reply:
I use try and except to handle this case, would you please take a review? Thx!